libs/core/state/src/operation/facade.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Action,
+  select,
+  Store,
+} from '@ngrx/store';
+import { Observable } from 'rxjs';
+
+import { DaffStoreFacade } from '../store/facade';
+import { DaffOperationStateSelectors } from './selectors';
+import { DaffOperationState } from './state';
+
+/**
+ * A facade for an operation state.
+ */
+export interface DaffOperationStateFacadeInterface<
+  TState extends DaffOperationState = DaffOperationState
+> extends DaffStoreFacade<Action> {
+  /**
+   * The loading state enum.
+   */
+  loadingState$: Observable<TState['loading']>;
+  /**
+   * Whether the operation state is in any of the loading states.
+   */
+  loading$: Observable<boolean>;
+  /**
+   * Whether the operation state is resolving.
+   */
+  resolving$: Observable<boolean>;
+  /**
+   * Whether the operation state is mutating.
+   */
+  mutating$: Observable<boolean>;
+  /**
+   * The errors in the operation state.
+   */
+  errors$: Observable<TState['errors']>;
+  /**
+   * Whether the operation state has any errors.
+   * If so, it should be considered to be in an "error" state.
+   */
+  hasErrors$: Observable<boolean>;
+}
+
+/**
+ * @inheritdoc
+ */
+export abstract class DaffOperationStateFacade<
```

Comment:
The abstract class here bloats the API, imo.

The interface itself is sound, but the rest of the code exists solely to reduce code but at the cost of encouraging inheritance, which I think will make usages more brittle.

I think we keep the interface, but delete the abstract class.

Reply:
agreed, removed